src/Vizkit3dWorld.cpp

```diff
@@ -330,4 +345,38 @@ void Vizkit3dWorld::applyCameraParams() {
     widget->getView(0)->getCamera()->setProjectionMatrixAsPerspective(osg::RadiansToDegrees(fovy), aspectRatio, zNear, zFar);
 }
 
+void Vizkit3dWorld::removeLinksChild(sdf::ElementPtr sdf_model, std::string child_element){
+
+    if (sdf_model->HasElement("link"))
```

Comment:
You don't need the two-stage if / while here. if the model does not have the element, linkElem will be NULL which is then ignored by while(). Just remove the if